data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.gateway/1.0/corda.p2p.gateway.json

```diff
@@ -5,24 +5,36 @@
   "description": "Configuration schema for the P2P gateway.",
   "type": "object",
   "properties": {
-    "hostAddress": {
-      "description": "The host name or IP address where the HTTP server will bind.",
-      "type": "string",
-      "format": "hostname",
-      "default": "0.0.0.0"
-    },
-    "hostPort": {
-      "description": "The HTTP port.",
-      "type": "integer",
-      "default": 8080,
-      "minimum": 0,
-      "maximum": 65535
-    },
-    "urlPath": {
-      "description": "The url path the gateway server will be listening to.",
-      "type": "string",
-      "default": "/",
-      "pattern": "^\\/[\\/.a-zA-Z0-9-]*$"
+    "servers": {
+      "type": "array",
+      "minItems": 1,
+      "description": "The HTTP servers. The pair hostAddress and hostPort must be unique for each server.",
+      "default": [{}],
```

Comment:
Have you tested this, validates as expected if you specify different levels of schema (I had to put in defaults on both levels for the link manager but this might be different)?  You can adapt the `ConfigurationValidationIntegrationTest`, to check manually without deploying.